### PR TITLE
Add networkMode to AKS 20200201 API

### PR DESCRIPTION
**Pull request by gossion**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by gossion** [comment]
@gtxistxgao FYI

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json [hunk 1]
```diff
@@ -1652,6 +1652,20 @@
           },
           "description": "Network policy used for building Kubernetes network."
         },
+        "networkMode": {
+          "type": "string",
+          "enum": [
+            "",
```
Comment (majastrz) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json]: >"" [](start = 12, length = 2)

What does empty string mean here?

Comment (gossion) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json]: @majastrz, this means that the networkMode is not set.

Comment (majastrz) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json]: Can we use a non-empty value to represent that? Something like `none` or `default` or 'NotSet`?

Comment (majastrz) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json]: Discussed over IM. The API is apparently already released, which violates the usual review workflow (swagger-first)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side given that the API is already released. Going forward, you need to get the swagger reviewed first before implementing it in the backend. This way feedback can actually be incorporated into the design of the API.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json (hunk 1 above)
Comment (gossion) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json]: thank you, will follow that.

**Comment by palma21** [comment]
Thanks, that seems reasonable, we'll do that moving forward. 

@ArcturusZhang let us know if anything else is needed before merging

**Comment by akning-ms** [comment]
@nullMDR can you help check why Python sdk is fail?

**Comment by NullMDR** [comment]
@changlong-liu @ArcturusZhang  Could you please help on the autorest failure?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json (hunk 1 above)
Comment (isra-fel) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json]: This is breaking C# generated code, see https://github.com/AzureSDKAutomation/azure-sdk-for-net/pull/418/files#diff-2f25f07072c1b2f687283e3233db4dc6R19

**Comment by ArcturusZhang** [comment]
> @NullMDR can you help check why Python sdk is fail?

Well, the root cause is that there is an empty string as enum value, which is breaking both go SDK and C# SDK generated code.

**Pull request merged**